messages/internationalization.mdx

```diff
@@ -1,14 +1,338 @@
+import { I18nextProvider } from 'react-i18next';
++ import { createI18n } from './i18n';
+
+export const LanguageContext = React.createContext(null);
+
+export const LanguageProvider = props => {
+  const { children } = props;
++ const [language, setLanguage] = React.useState('en');
++ const i18n = React.useMemo(() => createI18n(language), []);
+
+  return (
+-   <I18nextProvider i18n={}>
++   <I18nextProvider i18n={i18n}>
+-     <LanguageContext.Provider value={}>
++     <LanguageContext.Provider value={{ language }}>
+        {children}
+      </LanguageContext.Provider>
+    </I18nextProvider>
+  );
+};
+```
+
+Use `i18n.changeLanguage` when language changes:
+
+```diff
+import React from 'react';
+import { I18nextProvider } from 'react-i18next';
+  import { createI18n } from './i18n';
+
+export const LanguageContext = React.createContext(null);
+
+export const LanguageProvider = props => {
+  const { children } = props;
+  const [language, setLanguage] = React.useState('en');
+  const i18n = React.useMemo(() => createI18n(language), []);
+
++ const handleSetLanguage = newLanguage => {
++   i18n.changeLanguage(newLanguage);
++   setLanguage(newLanguage);
++ };
+
+  return (
+    <I18nextProvider i18n={i18n}>
+      <LanguageContext.Provider
+        value={{
+          language,
++         setLanguage: handleSetLanguage,
+        }}
+      >
+        {children}
+      </LanguageContext.Provider>
+    </I18nextProvider>
+  );
+};
+```
+
+## Global error message translation
+
+As we learned in previous section, we can customize error message globally using the `setErrorMessage` method provided by validators. So, we only need to provide the translated message when language changes:
+
+```javascript
+import { Validators } from '@lemoncode/fonk';
+import { createI18n } from './i18n';
+
+const i18n = createI18n('en');
+
+Validators.required.setErrorMessage(i18n.t('required'));
+Validators.email.setErrorMessage(i18n.t('email'));
+```
+
+If we are using `React`, we can use `useTranslation` hook and acces to the `t` method:
+
+```javascript
+import { Validators } from '@lemoncode/fonk';
+import { useTranslation } from 'react-i18next';
+
+
+const App = () => {
+  const { t } = useTranslation();
+  Validators.required.setErrorMessage(t('required'));
+  Validators.email.setErrorMessage(t('email'));
+  ...
+}
+```
+
+Check these `i18next React Final Form` examples:
+
+- [React React Final Form JS](https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/docs/messages/js/react-final-form/i18n-global-message).
+- [React React Final Form TS](https://codesandbox.io/s/github/lemoncode/fonk/tree/master/examples/docs/messages/ts/react-final-form/i18n-global-message).
+
+## Local error message translation
+
+This time, we will use `message` property available in `validationSchema`. Each time language is updated, we will provide the translated message to this property, that is, we need to refresh the form validation instance:
+
```

Comment:
I don't understand this paragraph

Reply:
Each time language selection is changed, we need to update value assigned to the message property in order to assign the new literal, we need to refresh the Form Validation instance